src/api/middleware/company.js

```diff
@@ -94,7 +94,32 @@ const isNotBlocked = (owner) => async (req, res, next) => {
             ValidationReasons.COMPANY_BLOCKED
         ));
     }
+    return next();
+};
 
+const isNotDisabled = (owner) => async (req, res, next) => {
+    const company = await (new CompanyService()).findById(owner, true);
+    if (company.isDisabled) {
+        return next(new APIError(
+            HTTPStatus.FORBIDDEN,
+            ErrorTypes.FORBIDDEN,
+            ValidationReasons.COMPANY_DISABLED
+        ));
+    }
+    return next();
+};
+
+const canModify = (companyId) => async (req, res, next) => {
```

Comment:
Still too generic, since this could be related to editing. Since it isn't, let's use `canToggleCompanyVisibility`. The Company "modifier" here helps understand that we are referring to the company's visibility and not to the ability of the company to modify visibility of its offers, for example